src/web/layouts/InteractiveImmersiveLayout.tsx

```diff
@@ -0,0 +1,329 @@
+import React from 'react';
```

Comment:
I would make the case that it might be worth abstracting out differences rather than copying `ImmersiveLayout.tsx`, can we not use `ImmersiveLayout` in the meantime?

Reply:
it just means that we will likely have to double the work if ever we need to make a change, and these files are big to maintain. I would prefer to include Immersive logic in this file

Reply:
Definitely, I don't think there is enough overlap with `ImmersiveLayout` relative to say other layouts (other than the name and nav) but I think we could (+ should) refactor all the *Layout files to pull out common things like nav handling and footer and simplify/re-use code a lot more.

(I think that should be a separate PR though as it will be large.)